src/autora/state/bundled.py

```diff
@@ -0,0 +1,175 @@
+           n  c
+        0  8  h
+        1  9  i
+
+        If they are passed without column names, no column names are inferred.
+        This is to ensure that accidental mislabeling of columns cannot occur.
+        Column names should usually be provided.
+        >>> dc4 = Delta(conditions=[(6,), (7,)])
+        >>> (s + dc4).conditions
+           0
+        0  6
+        1  7
+
+        Datatypes which are incompatible with a pd.DataFrame will throw an error:
+        >>> s + Delta(conditions="not compatible with pd.DataFrame")
+        Traceback (most recent call last):
+        ...
+        ValueError: ...
+
+        Experiment data can be updated using a Delta:
+        >>> ded1 = Delta(experiment_data=pd.DataFrame({"x": [1,2,3], "y": ["a", "b", "c"]}))
+        >>> (s + ded1).experiment_data
+           x  y
+        0  1  a
+        1  2  b
+        2  3  c
+
+        ... and are extended with each Delta:
+        >>> ded2 = Delta(experiment_data=pd.DataFrame({"x": [4, 5, 6], "y": ["d", "e", "f"]}))
+        >>> (s + ded1 + ded2).experiment_data
+           x  y
+        0  1  a
+        1  2  b
+        2  3  c
+        3  4  d
+        4  5  e
+        5  6  f
+
+        If they are passed without column names, no column names are inferred.
+        This is to ensure that accidental mislabeling of columns cannot occur.
+        >>> ded3 = Delta(experiment_data=pd.DataFrame([(7, "g"), (8, "h")]))
+        >>> (s + ded3).experiment_data
+           0  1
+        0  7  g
+        1  8  h
+
+        If there are already data present, the column names must match.
+        >>> (s + ded2 + ded3).experiment_data
+             x    y    0    1
+        0  4.0    d  NaN  NaN
+        1  5.0    e  NaN  NaN
+        2  6.0    f  NaN  NaN
+        3  NaN  NaN  7.0    g
+        4  NaN  NaN  8.0    h
+
+        `experiment_data` other than `pd.DataFrame` will be coerced into a `DataFrame` if possible.
+        >>> import numpy as np
+        >>> dc3 = Delta(
+        ...     experiment_data=np.core.records.fromrecords([(1, "a"), (2, "b")], names=["x", "y"]))
+        >>> (s + dc3).experiment_data
+           x  y
+        0  1  a
+        1  2  b
+
+        `experiment_data` which are incompatible with a pd.DataFrame will throw an error:
+        >>> s + Delta(experiment_data="not compatible with pd.DataFrame")
+        Traceback (most recent call last):
+        ...
+        ValueError: ...
+
+        `models` can be updated using a Delta:
+        >>> from sklearn.dummy import DummyClassifier
+        >>> dm1 = Delta(models=[DummyClassifier(constant=1)])
+        >>> dm2 = Delta(models=[DummyClassifier(constant=2), DummyClassifier(constant=3)])
+        >>> (s + dm1).models
+        [DummyClassifier(constant=1)]
+
+        >>> (s + dm1 + dm2).models
+        [DummyClassifier(constant=1), DummyClassifier(constant=2), DummyClassifier(constant=3)]
+
+        The last model is available under the `model` property:
```

Comment:
Hmm... how about `model` vs. `model_list` or `model_set`? Are those better than `models`?

Reply:
@musslick i'm fine with either of those! i'm also fine with `models` and `model` if others think it's not actually going to be an issue; just trying to maximally clear while still being Pythonic :)

Reply:
On reflection, I feel like I still prefer `models` rather than anything longer – it feels a bit more natural to me. Combined with the type annotations (like `BaseEstimator` vs `List[BaseEstimator]`) and type checking, I feel like the risk of accidental confusion is sufficiently small to be unproblematic. I'm open to persuasion there though.